dvc/api.py

```diff
@@ -92,6 +94,178 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
         return fd.read()
 
 
+def get_params(
```

Comment:
Until now, `dvc.api` has been about data-management/data-registry. If we have plans to add more API, it might make more sense to add this on a separate namespace: 
```python
from dvc.api.exp import get_params
```

Or,
```python
from dvc.api import Experiments

Experiments().params()
```

Reply:
I think that's a good point, the syntax for the second example looks a bit weird to me though

Reply:
@skshetry What args/kwargs would you expect `Experiments()` and `params()` to take?

Reply:
> Until now, dvc.api has been about data-management/data-registry. If we have plans to add more API, it might make more sense to add this on a separate namespace:
> from dvc.api.exp import get_params

I am not sure it makes sense _for now_ to add a separate namespace. If we extend the API "mirroring" the CLI, all commands are at the top-level namespace (`dvc {X}`), not only data management.
 
 > from dvc.api import Experiments
 
I think this is a good discussion to have but it requires more elaborated discussion and planning, which I believe is out of the scope of this P.R.

Reply:
> If we extend the API "mirroring" the CLI, all commands are at the top-level namespace (dvc {X}), not only data management.

I thought those porcelain APIs would exist in `dvc.porcelain` namespace. 

> I think this is a good discussion to have but it requires more elaborated discussion and planning, which I believe is out of the scope of this P.R.

I think it's a good time to discuss and choose the way we want to go. Once it's merged, we'll be stuck maintaining an API, or supporting duplicate APIs which might be more confusing.

Reply:
> I thought those porcelain APIs would exist in dvc.porcelain namespace.

What is wrong / what is the difference with the `dvc.api` namespace? 
Why `dvc.api.exp` makes more sense than the parent namespace?

I get the motivation of reusing git terminology but this case seems forced to me. 
I have only seen this term used to refer to a Python API in `dulwich` and our use case looks different, as dulwich explicitly exposes a lower-level public Python API.

> Until now, dvc.api has been about data-management/data-registry

Was this by design? Why name it `dvc.api` if the scope will be limited to data management?

> I think it's a good time to discuss and choose the way we want to go. Once it's merged, we'll be stuck maintaining an API, or supporting duplicate APIs which might be more confusing.

If we name this P.R. `params_show` and save it under `dvc.porcelain`, would still be an issue? Is this the concern based on naming? 

Names aside, the scope seems completely different to me. `Experiments` API as used in the snipped will require, IMO, far more effort to implement and reach consensus.

In addition, I think for some use cases `Experiments` will just be adding an unnecessary layer of complexity (for example, the original use case that motivates the P.R. "get parameters of the current stage")

> What if I write like this:
> 
> exp = Experiments()
> exp.params()
> exp.metrics()
> 
> Is it still weird?

We already use the term `experiments` in a very specific way in DVC and I don't really understand what `Experiments` will return; all `dvc experiments` in the current workspace?